src/tests/ftest/util/nvme_utils.py

```diff
@@ -283,11 +306,28 @@ def start_ior_load(self, storage='NVMe', operation="WriteRead",
         # Kill the server rank while IOR in progress
         if self.set_online_rebuild:
             time.sleep(30)
-            # Kill the server rank
-            if self.rank_to_kill is not None:
-                self.server_managers[0].stop_ranks([self.rank_to_kill], self.d_log, force=True)
-
-        # Wait to finish the thread
+            # Kill the server rank in BG thread
+            for _id, _rank in enumerate(self.rank_to_kill):
+                kill_rank_job.append(threading.Thread(target=self.kill_rank_thread,
+                                                     kwargs={"rank": _rank}))
+                kill_rank_job[_id].start()
+
+            # Kill the target from rank in BG thread
+            for _id, (key, value) in enumerate(self.pool_exclude.items()):
+                kill_target_job.append(threading.Thread(target=self.exclude_target_thread,
+                                                        kwargs={"rank": key,
+                                                                "target": value}))
```

Comment:
I hope we are not killing the same rank used in the above for loop (ie: which kills a rank). Probably, pool_exclude list doesn't contain rank_to_kill list...